react-hot-button-component/my-app/src/App.tsx

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useState } from 'react';
+import './App.css';
+import HotButton from './HotButton';
+
+function App() {
+  const [count, setCount] = useState(0);
+
+  function handleClick() {
+    setCount(count + 1);
+  }
+
+  function changeColor(count: number): any {
+    let color;
+    if (count < 3) {
+      color = 'cold';
+    } else if (count < 6) {
+      color = 'cool';
+    } else if (count < 9) {
+      color = 'tepid';
+    } else if (count < 12) {
+      color = 'warm';
+    } else if (count < 15) {
+      color = 'hot';
+    } else if (count < 18) {
+      color = 'nuclear';
+    } else {
+      color = 'cold';
+      setCount((count = 0));
```

Comment:
I don't understand this? It looks like you are trying to reset the count to 0 after 18 clicks, which is nice. The best way to do that is just `setCount(0);`.